internal/controller/api/hiveextension/v1beta1/agentclusterinstall_types.go

```diff
@@ -0,0 +1,179 @@
+package v1beta1
+
+import (
+	hivev1 "github.com/openshift/hive/apis/hive/v1"
+	corev1 "k8s.io/api/core/v1"
+	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+)
+
+// +genclient
+// +k8s:deepcopy-gen:interfaces=k8s.io/apimachinery/pkg/runtime.Object
+
+// AgentClusterInstall represents a request to provision an agent based cluster.
+//
+// +k8s:openapi-gen=true
+// +kubebuilder:subresource:status
+type AgentClusterInstall struct {
+	metav1.TypeMeta   `json:",inline"`
+	metav1.ObjectMeta `json:"metadata,omitempty"`
+
+	Spec   AgentClusterInstallSpec   `json:"spec"`
+	Status AgentClusterInstallStatus `json:"status,omitempty"`
+}
+
+// AgentClusterInstallSpec defines the desired state of the AgentClusterInstall.
+type AgentClusterInstallSpec struct {
+
+	// ImageSetRef is a reference to a ClusterImageSet. The release image specified in the ClusterImageSet will be used
+	// to install the cluster.
+	ImageSetRef hivev1.ClusterImageSetReference `json:"imageSetRef"`
+
+	// ClusterDeploymentRef is a reference to the ClusterDeployment associated with this AgentClusterInstall.
+	ClusterDeploymentRef corev1.LocalObjectReference `json:"clusterDeploymentRef"`
+
+	// ClusterMetadata contains metadata information about the installed cluster. It should be populated once the cluster install is completed. (it can be populated sooner if desired, but Hive will not copy back to ClusterDeployment until the Installed condition goes True.
+	ClusterMetadata *hivev1.ClusterMetadata `json:"clusterMetadata,omitempty"`
+
+	// ManifestsConfigMapRef is a reference to user-provided manifests to
+	// add to or replace manifests that are generated by the installer.
+	ManifestsConfigMapRef *corev1.LocalObjectReference `json:"manifestsConfigMapRef,omitempty"`
+
+	// Networking is the configuration for the pod network provider in
+	// the cluster.
+	Networking Networking `json:"networking"`
+
+	// SSHPublicKey will be added to all cluster hosts for use in debugging.
+	// +optional
+	SSHPublicKey string `json:"sshPublicKey,omitempty"`
+
+	// ProvisionRequirements defines configuration for when the installation is ready to be launched automatically.
+	ProvisionRequirements ProvisionRequirements `json:"provisionRequirements"`
+
+	// ControlPlane is the configuration for the machines that comprise the
+	// control plane.
+	// +optional
+	ControlPlane *AgentMachinePool `json:"controlPlane,omitempty"`
+
+	// Compute is the configuration for the machines that comprise the
+	// compute nodes.
+	// +optional
+	Compute []AgentMachinePool `json:"compute,omitempty"`
```

Comment:
What is the key? it's just a list of "enabled, disabled..."

Reply:
Good catch, should probably have the assumed name of from the installer

Reply:
@dgoodwin i don't know what should be the key here. How users identify the compute nodes?

Reply:
I was working off the installer installconfig structure for this, so we could use the same key they did: https://github.com/openshift/installer/blob/master/pkg/types/machinepools.go#L44

They default that to "worker" for the main compute pool.

Reply:
We will handle it in different ticket. I opened https://issues.redhat.com/browse/MGMT-6022

Reply:
If the key is master/worker then we don't need `ControlPlane *AgentMachinePool `json:"controlPlane,omitempty"``